include/openssl/bio.h.in

```diff
@@ -885,34 +885,21 @@ void BIO_copy_next_retry(BIO *b);
  * long BIO_ghbn_ctrl(int cmd,int iarg,char *parg);
  */
 
-# define ossl_bio__attr__(x)
-# if defined(__GNUC__) && defined(__STDC_VERSION__) \
-    && !defined(__MINGW32__) && !defined(__MINGW64__) \
+# if defined(__GNUC__) && !defined(__MINGW32__) && !defined(__MINGW64__) \
     && !defined(__APPLE__)
-    /*
-     * Because we support the 'z' modifier, which made its appearance in C99,
-     * we can't use __attribute__ with pre C99 dialects.
-     */
-#  if __STDC_VERSION__ >= 199901L
-#   undef ossl_bio__attr__
-#   define ossl_bio__attr__ __attribute__
-#   if __GNUC__*10 + __GNUC_MINOR__ >= 44
-#    define ossl_bio__printf__ __gnu_printf__
-#   else
-#    define ossl_bio__printf__ __printf__
-#   endif
-#  endif
+#  define ossl_bio__attr__ __attribute__
+# else
+#  define ossl_bio__attr__(x)
 # endif
 int BIO_printf(BIO *bio, const char *format, ...)
-ossl_bio__attr__((__format__(ossl_bio__printf__, 2, 3)));
+    ossl_bio__attr__((__format__(printf, 2, 3)));
 int BIO_vprintf(BIO *bio, const char *format, va_list args)
-ossl_bio__attr__((__format__(ossl_bio__printf__, 2, 0)));
+    ossl_bio__attr__((__format__(printf, 2, 0)));
 int BIO_snprintf(char *buf, size_t n, const char *format, ...)
-ossl_bio__attr__((__format__(ossl_bio__printf__, 3, 4)));
+    ossl_bio__attr__((__format__(printf, 3, 4)));
```

Comment:
We saw that these didn't really work on some platforms. Maybe this diff fixes it now since it becomes a lot simpler.